signing_service/signing_service/signing_service.py

```diff
@@ -211,10 +255,40 @@ def __call__(self, parser, namespace, values, option_string=None):
         action=make_secret_provider_factory(env_variable_name='ETHEREUM_PRIVATE_KEY', base64_convert=True),
         help='Ethereum private key for Singing Service.',
     )
-    parser.set_defaults(ethereum_private_key=b64decode(os.environ.get('ETHEREUM_PRIVATE_KEY')) if os.environ.get('ETHEREUM_PRIVATE_KEY') is not None else None)  # type: ignore
+    parser.set_defaults(
+        ethereum_private_key=b64decode(os.environ.get('ETHEREUM_PRIVATE_KEY'))  # type: ignore
+        if os.environ.get('ETHEREUM_PRIVATE_KEY') is not None
+        else None
+    )
+
+    signing_service_private_key_parser_group = parser.add_mutually_exclusive_group()
+    signing_service_private_key_parser_group.add_argument(
+        '--signing-service-private-key',
+        dest='signing_service_private_key',
+        action=make_secret_provider_factory(read_command_line=True, base64_convert=True),
+        help='Singing Service private key.',
+    )
+    signing_service_private_key_parser_group.add_argument(
+        '--signing-service-private-key-path',
+        dest='signing_service_private_key',
+        action=make_secret_provider_factory(use_file=True, base64_convert=True),
+        help='Singing Service private key.',
+    )
+    signing_service_private_key_parser_group.add_argument(
+        '--signing-service-private-key-from-env',
+        dest='signing_service_private_key',
+        action=make_secret_provider_factory(env_variable_name='SIGNING_SERVICE_PRIVATE_KEY', base64_convert=True),
+        help='Singing Service private key.',
+    )
+    parser.set_defaults(
+        signing_service_private_key=b64decode(os.environ.get('SIGNING_SERVICE_PRIVATE_KEY'))  # type: ignore
```

Comment:
hm... conditional for default doesn't seem to be a good idea...

Reply:
Removes as discussed in https://github.com/golemfactory/concent/pull/688#discussion_r208591838.